app/assets/javascripts/app/views/notifications_view.js

```diff
@@ -1,94 +1,78 @@
     var note = $(evt.target).closest(".stream-element");
     var unread = note.hasClass("unread");
     var guid = note.data("guid");
-    if (unread){ this.setRead(guid); }
-    else { this.setUnread(guid); }
+    if (unread) {
+      this.collection.setRead(guid);
+    } else {
+      this.collection.setUnread(guid);
+    }
   },
 
-  getAllUnread: function() { return $(".media.stream-element.unread"); },
-
-  setRead: function(guid) { this.setUnreadStatus(guid, false); },
-
-  setUnread: function(guid){ this.setUnreadStatus(guid, true); },
-
-  setUnreadStatus: function(guid, state){
-    $.ajax({
-      url: "/notifications/" + guid,
-      data: { set_unread: state },
-      type: "PUT",
-      context: this,
-      success: this.clickSuccess
-    });
+  markAllRead: function() {
+    this.collection.setAllRead();
   },
 
-  clickSuccess: function(data) {
-    var guid = data.guid;
-    var type = $(".stream-element[data-guid=" + guid + "]").data("type");
-    this.updateView(guid, type, data.unread);
-  },
+  onChagedUnreadStatus: function(model) {
+    var guid = model.getGUID();
+    var unread = model.get("unread");
+    var translationKey = unread ? "notifications.mark_read" : "notifications.mark_unread";
+    var note = $(".stream-element[data-guid=" + guid + "]");
 
-  markAllRead: function(evt){
-    if(evt) { evt.preventDefault(); }
-    var self = this;
-    this.getAllUnread().each(function(i, el){
-      self.setRead($(el).data("guid"));
-    });
-  },
+    note.find(".entypo-eye")
+      .tooltip("destroy")
+      .removeAttr("data-original-title")
+      .attr("title", Diaspora.I18n.t(translationKey))
+      .tooltip();
 
-  updateView: function(guid, type, unread) {
-    var change = unread ? 1 : -1,
-        allNotes = $("#notifications_container .list-group > a:eq(0) .badge"),
-        typeNotes = $("#notifications_container .list-group > a[data-type=" + type + "] .badge"),
-        headerBadge = $(".notifications-link .badge"),
-        note = $(".notifications .stream-element[data-guid=" + guid + "]"),
-        markAllReadLink = $("a#mark_all_read_link"),
-        translationKey = unread ? "notifications.mark_read" : "notifications.mark_unread";
-
-    if(unread){ note.removeClass("read").addClass("unread"); }
+    if (unread) { note.removeClass("read").addClass("unread"); }
     else { note.removeClass("unread").addClass("read"); }
+  },
 
-    $(".unread-toggle .entypo-eye", note)
-        .tooltip("destroy")
-        .removeAttr("data-original-title")
-        .attr("title",Diaspora.I18n.t(translationKey))
-        .tooltip();
-
-    [allNotes, typeNotes, headerBadge].forEach(function(element){
-      element.text(function(i, text){
-        return parseInt(text) + change;
-      });
-    });
+  updateView: function() {
+    var notificationsContainer = $("#notifications_container");
 
-    [allNotes, typeNotes].forEach(function(badge) {
-      if(badge.text() > 0) {
+    var updateBadge = function(badge, count) {
```

Comment:
You assigned the function to a variable inside oft a function. I suggested to make this a function of the view.

One might want to use the function later for something else and one would be able to test the function.